scripts/repo_rules.sh

```diff
@@ -1,94 +1,260 @@
 #!/usr/bin/env bash
-# Script to validate repo rules for paketo cnbs
-# usage e.g.
-# GITHUB_TOKEN=$(cat /path/to/github_token) repo_rules.sh paketo-buildpacks/php-dist
-# Pass second argument --verbose to print the api output json
-# You can get a personal token from github.com/settings. Token just requires repo perm
-
-set -e
-set -u
+
+set -eu
 set -o pipefail
 
-if [  $# -lt 1 ] || [  -z "${GITHUB_TOKEN:-}" ]; then
-    echo "usage: GITHUB_TOKEN=<token> $0 <org>/<repo> [--verbose]"; exit 1;
-fi
-
-if ! [[ "$1" =~ [a-z-]+/[a-z-]+ ]]; then
-    echo Provide a valid argument of syntax "<org>/<repo>"
-    exit 1
-fi
-
-user=${1%%/*}
-repo=${1##*/}
-green="\\0033[0;32m"
-red="\\0033[0;31m"
-reset="\\0033[0;39m"
-ret=0
-
-bad() {
-    echo -e "- $red""$1""$reset"
-    ret=1
+readonly PROGDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# shellcheck source=SCRIPTDIR/.util/print.sh
+source "${PROGDIR}/.util/print.sh"
+
+function main() {
+  local repo token branch verbose
+  while [[ "${#}" != 0 ]]; do
+    case "${1}" in
+      --repo)
+        repo="${2}"
+        shift 2
+        ;;
+
+      --token)
+        token="${2}"
+        shift 2
+        ;;
+
+      --branch)
+        branch="${2}"
+        shift 2
+        ;;
+
+      --verbose)
+        verbose=true
+        shift 1
+        ;;
+
+      --help|-h)
+        shift 1
+        usage
+        exit 0
+        ;;
+
+      "")
+        # skip if the argument is empty
+        shift 1
+        ;;
+
+      *)
+        util::print::error "unknown argument \"${1}\""
+    esac
+  done
+
+  if [[ -z "${repo:-}" ]]; then
+    usage
+    echo
+    util::print::error "--repo is a required flag"
+  fi
+
+  if [[ -z "${token:-}" ]]; then
+    usage
+    echo
+    util::print::error "--token is a required flag"
+  fi
+
+  if [[ -z "${branch:-}" ]]; then
+    branch="main"
+  fi
+
+  if [[ ! "${repo}" =~ [a-z-]+/[a-z-]+ ]]; then
+    util::print::error "--repo argument must match <org>/<name> format"
+  fi
+
+  if rules "${token}" "${repo}" "${branch}" "${verbose}"; then
+    util::print::success "Valid"
+  else
+    util::print::error "Invalid"
+  fi
+}
+
+function usage() {
+  cat <<-USAGE
+repo_rules.sh --repo <repo> --token <token> [OPTIONS]
+
+Validates branch protection rules for a GitHub repository.
+
+OPTIONS
+  --branch <branch>  branch to check for protection rules (default: main)
+  --help  -h         prints the command usage
+  --repo <repo>      name of the GitHub repository to check in the form <org>/<name>
+  --token <token>    GitHub token used to check the repository
+USAGE
```

Comment:
Does --verbose need to be one of the options listed in the --help output also?

Reply:
Ah. Yeah. I missed that.